core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -1485,7 +1478,10 @@ public synchronized Sha256Hash hashForWitnessSignature(
 
     @Override
     public int messageSize() {
-        boolean useSegwit = hasWitnesses() && allowWitness(protocolVersion);
+        return messageSize(hasWitnesses() && allowWitness(protocolVersion));
```

Comment:
I kinda miss the `useSegwit` name here. Maybe we could add a comment to help out the reader?

Reply:
The line now looks like this:

`return messageSize(useSegwitSerialization());`

Ok?